Code/Tools/BundleLauncher/O3DE_SDK_Launcher.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Component/ComponentApplication.h>
+#include <AzCore/Memory/SystemAllocator.h>
+#include <AzCore/Settings/SettingsRegistryMergeUtils.h>
+#include <AzCore/Utils/Utils.h>
+#include <AzFramework/Process/ProcessWatcher.h>
+
+#include <cstdlib>
+#include <mach-o/dyld.h>
+
+int main(int argc, char* argv[])
+{
+    // We need to pass in the engine path since we won't be able to find it by searching upwards.
+    // We can't use any containers that use our custom allocator till after the call to ComponentApplication::Create()
+    AZ::IO::FixedMaxPath processPath = AZ::Utils::GetExecutableDirectory();
+    AZ::IO::FixedMaxPath enginePath = (processPath / "../Engine").LexicallyNormal();
+    auto enginePathParam = AZ::SettingsRegistryInterface::FixedValueString::format(R"(--engine-path="%s")", enginePath.c_str());
+    // Uses the fixed_vector deduction guide to determine the type is AZStd::fixed_vector<char*, 3>
+    AZStd::fixed_vector commandLineParams{ processPath.Native().data(), enginePathParam.data(), static_cast<char*>(nullptr) };
+
+
+    
+    // Create a ComponentApplication to initialize the AZ::SystemAllocator and initialize the SettingsRegistry
+    AZ::ComponentApplication application(2, commandLineParams.data());
+    application.Create(AZ::ComponentApplication::Descriptor());
```

Comment:
Between the ComponentApplication constructor and the call the `Create` is the place where the specialization tag of `o3de_sdk` can be injected into the Settings Registry to allow the O3DE_SDK launcher application to load any setreg files specific to it. Such as `O3DE_SDK_InstallPath.o3de_sdk.setreg`.
This will stop other applications such as the AssetProcessor and Editor from attempting to load in that settings registry file.

The [MergeSettingsToRegistry_AddBuildSystemTargetSpecialization](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.h#L108-L110) function can be used here

```suggestion
    AZ::ComponentApplication application(commandLineParams.size(), commandLineParams.data());
    AZ::SettingsRegistryMergeUtils::MergeSettingsToRegistry_AddBuildSystemTargetSpecialization(*AZ::SettingsRegistry::Get(), "o3de_sdk");
    application.Create(AZ::ComponentApplication::Descriptor());
```

Reply:
Fixed.